catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -2482,73 +2418,45 @@ protected ResourceResponse getResource(ResourceRequest resourceRequest, boolean
-                    LOGGER.info(
-                            "Unable to get resource from cache. Have to retrieve it from source {}",
-                            resourceSourceName);
-                }
-            }
-
-            if (resourceResponse == null) {
-                // retrieve product from specified federated site if not in cache
-                if (!resourceSourceName.equals(getId())) {
-                    LOGGER.debug("Searching federatedSource {} for resource.", resourceSourceName);
-                    LOGGER.debug("metacard for product found on source: {}", resolvedSourceId);
-                    FederatedSource source;
+            // retrieve product from specified federated site if not in cache
+            if (!resourceSourceName.equals(getId())) {
+                LOGGER.debug("Searching federatedSource {} for resource.", resourceSourceName);
+                LOGGER.debug("metacard for product found on source: {}", resolvedSourceId);
+                FederatedSource source;
 
-                    source = frameworkProperties.getFederatedSources()
-                            .get(resourceSourceName);
-                    if (source != null) {
-                        LOGGER.debug("Adding federated site to federated query: {}",
-                                source.getId());
-                    }
+                source = frameworkProperties.getFederatedSources()
+                        .get(resourceSourceName);
+                if (source != null) {
+                    LOGGER.debug("Adding federated site to federated query: {}",
+                            source.getId());
+                }
 
-                    if (source != null) {
-                        LOGGER.debug("Retrieving product from remote source {}", source.getId());
-                        ResourceRetriever retriever = new RemoteResourceRetriever(source,
-                                responseURI,
-                                requestProperties);
-                        try {
-                            resourceResponse =
-                                    frameworkProperties.getReliableResourceDownloadManager()
-                                            .download(resourceRequest, metacard, retriever);
-                        } catch (DownloadException e) {
-                            LOGGER.info("Unable to download resource", e);
-                        }
-                    } else {
-                        LOGGER.warn("Could not find federatedSource: {}", resourceSourceName);
-                    }
-                } else {
-                    LOGGER.debug("Retrieving product from local source {}", resourceSourceName);
-                    ResourceRetriever retriever =
-                            new LocalResourceRetriever(frameworkProperties.getResourceReaders(),
-                                    responseURI,
-                                    requestProperties);
+                if (source != null) {
```

Comment:
Move the debug statement from above into this block. There's no reason to check if source is null twice.